pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -211,7 +208,15 @@ func (proxy *DataSourceProxy) getDirector() func(req *http.Request) {
 		}
 
 		if proxy.ds.JsonData != nil && proxy.ds.JsonData.Get("oauthPassThru").MustBool() {
-			addOAuthPassThruAuth(proxy.ctx, req)
+			token, err := social.GetCurrentOAuthToken(proxy.ctx.Req.Context(), *proxy.ctx.SignedInUser)
+			if err != nil {
+				logger.Error("Error fetching oauth token for user", "error", err)
+				return
+			}
+			if token != nil {
+				req.Header.Del("Authorization")
+				req.Header.Add("Authorization", fmt.Sprintf("%s %s", token.Type(), token.AccessToken))
+			}
```

Comment:
If `err` is nil then it's very fair to assume that `token` is not.

```suggestion
			req.Header.Del("Authorization")
			req.Header.Add("Authorization", fmt.Sprintf("%s %s", token.Type(), token.AccessToken))
```